widget/list.go

```diff
@@ -512,24 +316,95 @@ func (l *listLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
 	return fyne.NewSize(0, 0)
 }
 
-func (l *listLayout) getObjects() []fyne.CanvasObject {
-	objects := l.children
-	objects = append(objects, l.separators...)
-	return objects
+func (l *listLayout) getItem() *listItem {
+	item := l.itemPool.Obtain()
+	if item == nil {
+		if f := l.list.CreateItem; f != nil {
+			item = newListItem(f(), nil)
+		}
+	}
+	return item.(*listItem)
+}
+func (l *listLayout) offsetUpdated(pos fyne.Position) {
+	if l.list.offsetY == pos.Y {
+		return
+	}
+	l.list.offsetY = pos.Y
+	l.updateList()
 }
 
-func (l *listLayout) appendedItem(objects []fyne.CanvasObject) {
-	if len(objects) > 1 {
-		objects[len(objects)-1].Move(fyne.NewPos(0, objects[len(objects)-2].Position().Y+l.list.itemMin.Height+theme.SeparatorThicknessSize()))
-	} else {
-		objects[len(objects)-1].Move(fyne.NewPos(0, 0))
+func (l *listLayout) setupListItem(item fyne.CanvasObject, id ListItemID) {
+	li := item.(*listItem)
+	previousIndicator := li.selected
+	li.selected = false
+	for _, s := range l.list.selected {
+		if id == s {
+			li.selected = true
```

Comment:
If this is setting up a single list item, wouldn't it be more efficient to break the loop when the list item selected was found?